packages/lib/models/Setting.ts

```diff
@@ -2924,6 +2924,19 @@ class Setting extends BaseModel {
 		if (name === 'cli') return 'CLI';
 		return name[0].toUpperCase() + name.substr(1).toLowerCase();
 	}
+
+	public static async resaveSecureSettings() {
+		const keys = this.keys(false, null, { secureOnly: true });
+
+		for (const key of keys) {
+			const value = this.value(key);
+			if (value) {
+				await this.keychainService().deletePassword(`setting.${key}`);
+				this.changedKeys_.push(key);
+				await this.saveAll();
```

Comment:
To-do: Move to keychainservicedriver — only delete after adding to the new keychain to prevent data loss.